### PR TITLE
(WIP) First stab at ReadMe with local & Colab instructions

### PR DESCRIPTION
**Pull request by arnaudsj**
Hi Jhenna,

Here is a first stab at the ReadMe. Please take a look and please try to follow the instructions to make sure I did not miss anything. I am going to try to add DeepNote and Paperspace instructions as well (just using the public Github Repo as well) n order to give enough options to the students.

Let me know if you run into any issues.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,51 @@
-# Mini-Projects
+# Machine Learning Engineering Course Mini-Projects
+
+This is the repository of the current miniprojects for Springboard's Machine Learning Engineering course. As you work through the curriculum, you will need to complete each and everyone of those mini-projects and review and discuss them with your mentor.
```
Comment (jvoorhis-sb) [inline README.md]: @arnaudsj can you remove Springboard name from this? and just say "This is the repository of the current miniprojects for the Machine Learning Engineering course."

File: README.md [hunk 2]
```diff
@@ -1,5 +1,51 @@
...
+
+## Installation
+
+Below are details instructions on how to get those mini projects up and running. None of those projects should require tons of CPU or memory, so you are welcome to either run those locally on your computer, or through any hosted jupyter notebook services such as Deepnote or Google's Collab.
+
+### Local
+
+To run those projects locally, we recommend you first install Anaconda for your operating system (Windows, Mac, Linux) by downloading it and following the instructions here: https://www.anaconda.com/products/individual#Downloads
+
+Once anaconda is installed, run the following command to create dedicated python environment with all the libraries and packages needed to run all the mini projects in this repo:
```
Comment (jvoorhis-sb) [inline README.md]: @arnaudsj a student needs to first make sure they are in this repos root, can you add instruction for them to clone their fork to their local machine?

File: README.md [hunk 3]
```diff
@@ -1,5 +1,51 @@
...
+
+Then create an ipython kernel for Jupyter based on this environment:
+```python -m ipykernel install --user --name mec-miniprojects --display-name "Python (MEC Mini-Projects)"```
+
+Finally, launch Jupyter Lab and navigate to each mini-project notebook and make sure to select the kernel that was created above:
+```jupyter lab```
+
+### Google Colab
+
+It is actually very easy to work and run the notebooks for each mini-projects in Google Colab by leveraging the Github integration. Please note that you will need to have forked the mini-projects repository for this to work.
```
Comment (jvoorhis-sb) [inline README.md]: @arnaudsj can you also add a note that this only works for mini-projects that are in Jupyter notebooks. Some of the projects are in markdown files and cannot be imported to colab this way. (alternatively we could update those files to just be in notebooks which might be preferable)

File: README.md [hunk 4]
```diff
@@ -1,5 +1,51 @@
...
+- Once the authorization processed, you should now see a list of all your repositories on Github. All you need to do for this step is to select the forked mec-mini-projects repo
+- Once the mec-mini-projects repo selected Colab will ask you which notebook you want to open in that repo. 
+- After selecting the mini-project notebook of your choice, it will be open and ready to use
+
+Once you are done working on your mini-project notebook in Colab, you can easily save a copy of your notebook back to Github by clicking on File -> Save a Copy To Github, it will ask you to enter a commit message, and a file name, and will then commit your changes and work to Github.
+
+#### Cautionary words
+Google Collab, while being super easy to get up and running and a very capable environment for all the MEC mini-projects, does have one main drawback: python library versions. You are able to easily install packages, but Colab comes with a fleet of existing datascience related packages, and they may not always be the same version that is expected in the various mini-projects. There can be sometimes some incompatibilities, requiring you to install or upgrade some of the packages needed.
```
Comment (jvoorhis-sb) [inline README.md]: @arnaudsj typo in the last sentence --> "There can sometimes be incompatibilities..."

File: README.md [hunk 5]
```diff
@@ -1,5 +1,51 @@
...
+
+## Contributions & PRs welcome!
+
+With the objective to proviide the best possible learning content to our students, please do note hesitate to open issues in Github or submit PRs if you catch any typos or issues in those mini-projects.
```
Comment (jvoorhis-sb) [inline README.md]: @arnaudsj typo in "provide" there is an extra "i"

File: environment.yml [hunk 6]
```diff
@@ -1,5 +1,5 @@
 
-name: springboard-mec
+name: mec-miniprojects
```
Comment (jvoorhis-sb) [inline environment.yml:2]: Thank you for making this change!

**Review by jvoorhis-sb — CHANGES_REQUESTED** [review]
I think this is a great start. In addition to the changes below, we also discussed adding some additional support for how students should manage their forked repo. I am okay with those supporting pieces coming later and being able to get the local and colab solutions out to students as quickly as possible.

File: README.md (hunk 1 above)
Comment (arnaudsj) [reply README.md]: Got it!

File: README.md (hunk 2 above)
Comment (arnaudsj) [reply README.md]: Got it. Added a new paragraph with a link to the excellent Github Doc: https://docs.github.com/en/github/getting-started-with-github/fork-a-repo

File: README.md (hunk 3 above)
Comment (arnaudsj) [reply README.md]: Got it! Added a small precision, and yes, let's put this on the list to convert this one to a Notebook!

File: README.md (hunk 4 above)
Comment (arnaudsj) [reply README.md]: Got it. Thank you for catching it!

**Pull request merged**